realm/src/main/java/io/realm/internal/Table.java

```diff
@@ -1440,6 +1443,9 @@ public long findFirstDate(long columnIndex, Date date) {
 
     @Override
     public long findFirstString(long columnIndex, String value) {
+        if (value == null) {
```

Comment:
Yes, and even with null support, we still need a check when the field is not-nullable.